### PR TITLE
Page Content Focus: Add welcome guides

### LINKED ISSUE #51822: Content editing in site editor guidance
## What problem does this address?
Provide some early guidance for users who are yet to experience content editing in the site editor.

## What is your proposed solution?
Re-use the welcome modal in two different scenarios.

**A user first hits the site editor with a page in context (e.g. after clicking a page in the pages sidebar)**

https://github.com/WordPress/gutenberg/assets/1072756/8396bc0f-d69d-4d96-afb9-68db0bdc3662

**A user first switches to editing a template from a page**

https://github.com/WordPress/gutenberg/assets/1072756/7c01cfc9-283d-4615-8e1a-9048b13b1beb

The video files are attached below. We may need to convert these to gif .

https://github.com/WordPress/gutenberg/assets/1072756/64a46ea9-85f7-4ef3-b8fe-c75c14825135

https://github.com/WordPress/gutenberg/assets/1072756/0621c794-ce40-4837-b076-054e1161eb86

@WordPress/gutenberg-design

### PR DESCRIPTION
**Pull request by noisysocks**
## What?
Closes https://github.com/WordPress/gutenberg/issues/51822.

Adds welcome guides which appear to the user when they open a page in the site editor and when they edit that page's template.

## How?
- New `WelcomeGuidePage` and `WelcomeGuideTemplate` components in `@wordpress/edit-site` do the bulk of the lifting.
- Supporting changes to `WelcomeGuide` in `@wordpress/components` which is what all of our guides use.
    - Use default button styling for the Previous and Next buttons instead of (buggy!) custom styling.
    - Use default accent/theme colour for Previous button, Next button, and current page indicator instead of a custom hardcoded blue.
    - Do not show X button if there is only one page.

## Testing Instructions
1. Go to Appearance → Editor → Pages and select a page to edit.
2. A guide should appear.
3. Click _Edit template_ in the sidebar.
4. Another guide should appear.
5. Check that exiting the guides in the page editor, site editor, widgets editor, and widgets customiser still work.

Dismissing the guides should prevent them from appearing again.

For testing purposes, you can run this in DevTools to re-enable the new guides:

```
wp.data.dispatch( 'core/preferences' ).toggle( 'core/edit-site', 'welcomeGuidePage' );
wp.data.dispatch( 'core/preferences' ).toggle( 'core/edit-site', 'welcomeGuideTemplate' );
```

## Screenshots or screencast 

The new guides:

https://github.com/WordPress/gutenberg/assets/612155/e1c3bedd-5ae1-4f27-a14b-0e5e0db72ce7

The old guides:

| Before | After |
| - | - | 
| Screenshot 2023-06-28 at 16 56 16 | Screenshot 2023-06-28 at 16 55 58 |

### CODE REVIEW

File: editing-your-page.mp4
Comment (noisysocks) [inline editing-your-page.mp4:1]: Will upload this file to s.w.org before merging.

File: editing-your-template.mp4
Comment (noisysocks) [inline editing-your-template.mp4:1]: Will upload this file to s.w.org before merging.

**Comment by carolinan** [comment]
I closed both welcome guides after watching them the first time.
Now when I am editing the page and I click on Options, Welcome guide, I see the "Edit Site" welcome guide, not the page welcome guide.

**Comment by noisysocks** [comment]
> I closed both welcome guides after watching them the first time.
> Now when I am editing the page and I click on Options, Welcome guide, I see the "Edit Site" welcome guide, not the page welcome guide.

@SaxonF What's expected here?

**Comment by SaxonF** [comment]
> What's expected here?

There is still value in a more general welcome guide outside of learning about switching to/from pages and templates which is more contextual. I think we should setup a follow-up issue that brings multiple steps into the site editor welcome guide including education around pages and templates in more abstract form like @jameskoster and @joen suggested in the original issue (similar to post editor).

**Review by SaxonF — APPROVED** [review]
Works well in testing.

**Review by talldan — APPROVED** [review]
LGTM!

**Comment by noisysocks** [comment]
Just waiting to get the video files uploaded to s.w.org before merging.

https://meta.trac.wordpress.org/ticket/7113

**Comment by tellthemachines** [comment]
Removed the label as this was included in the package updates for Beta 3 yesterday.

**Comment by afercia** [comment]
> Do not show X button if there is only one page.

This change has an impact on accessibility. Please mark any PR that changes accessibility-related stuff (like the the Modal dialog initial focus in this case) with the accessibility label.

It is certainly possible to not show the X button, although I'd prefer to always show it to provide a consistent user experience. However, when removing the X button _and_ initial focus goes to the _last_ element within the modal dialog, all the previous content gets skipped by screen readers. That's less than ideal, as screen reader users will have no clue there's some content before the 'Get started' button.
They will only hear, for example:

`Welcome to the template editor` (the modal dialog aria-label)
`Get started`

In this cases, it would be better to set initial focus on the modal dialog itself. Will create a new issue.

**Comment by noisysocks** [comment]
Thanks @afercia! Good spotting.

**Pull request merged**